controllers/drplacementcontrol.go

```diff
@@ -800,6 +814,29 @@ func (d *DRPCInstance) createVRGManifestWork(homeCluster string) error {
 	return nil
 }
 
+func (d *DRPCInstance) ensureNamespaceExistsOnManagedCluster(homeCluster string) error {
+	// verify namespace exists on target cluster
+	namespaceExists, err := d.namespaceExistsOnManagedCluster(homeCluster)
+
+	d.log.Info(fmt.Sprintf("createVRGManifestWork: namespace '%s' exists on cluster %s: %t",
+		d.instance.Namespace, homeCluster, namespaceExists))
+
+	if err != nil {
+		return fmt.Errorf("failed to verify if namespace '%s' on cluster %s exists: %w",
+			d.instance.Namespace, homeCluster, err)
+	}
+
```

Comment:
(nit) check if namespaceExists first and return nil, avoids a bigger if block as in golang [practices](https://golang.org/doc/effective_go#if). Just a stylistic comment.